samples/detachSubscription.js

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * This sample demonstrates how to detach subscriptions with the
+ * Google Cloud Pub/Sub API.
+ *
+ * For more information, see the README.md under /pubsub and the documentation
+ * at https://cloud.google.com/pubsub/docs.
+ *
+ * Note: This sample is NOT currently included in the system-test, because
+ * the feature isn't enabled on the production server.
+ */
+
+'use strict';
+
+// sample-metadata:
+//   title: Detach Subscription
+//   description: Detaches a subscription from a topic.
+//   usage: node detachSubscription.js <existing-subscription-name>
+
+function main(subscriptionName = 'YOUR_EXISTING_SUBSCRIPTION_NAME') {
+  // [START pubsub_detach_subscription]
+  /**
+   * TODO(developer): Uncomment these variables before running the sample.
+   */
+  // const subscriptionName = 'YOUR_EXISTING_SUBSCRIPTION_NAME';
+
+  // Imports the Google Cloud client library
+  const {PubSub} = require('@google-cloud/pubsub');
+
+  // Creates a client; cache this for further use
+  const pubSubClient = new PubSub();
+
+  async function detachSubscription() {
+    // Gets the status of the existing subscription
+    const sub = pubSubClient.subscription(subscriptionName);
+    const [detached] = await sub.detached();
+    console.log(
+      `Subscription ${subscriptionName} 'before' detached status: ${detached}`
+    );
+
+    await pubSubClient.detachSubscription(subscriptionName);
+    console.log(`Subscription ${subscriptionName} detach request was sent.`);
+
+    const [updatedDetached] = await sub.detached();
+    console.log(
+      `Subscription ${subscriptionName} 'after' detached status: ${updatedDetached}`
+    );
+  }
+
+  detachSubscription();
+  // [END pubsub_detach_subscription]
+}
+
+process.on('unhandledRejection', err => {
```

Comment:
I believe this is the stuff we're looking for now:
```js

main(...process.argv.slice(2)).catch(err => {
  console.error(err.message);
  process.exitCode = 1;
});
process.on('unhandledRejection', (err) => {
  console.error(err.message);
  process.exitCode = 1;
});
```

Reply:
So, I'm slightly concerned there is Zalgo in that ordering. (This objection is a bit ironic since I helped author that sample style update. >_>) That can be a separate discussion though.

Reply:
Ugh.  @bcoe @sofisl @fhinkel